stores/healthcareProfessionalsStore.ts

```diff
@@ -89,18 +87,21 @@ export const useHealthcareProfessionalsStore = defineStore(
 
             const facilitiesForRelationshipCreationArray = selectedFacilities.value
 
+            // This is the array to be sent to the backend if there is a change in the relations
+            let facilitiesRelationsToSelectedHealthcareProfessional: Relationship[] = []
+
             if (facilitiesForRelationshipCreationArray.length) {
-                //For each is used here to only add the necessary actions to the already created ref array of relationships
-                facilitiesForRelationshipCreationArray.forEach(facility => createFacilitiesRelationArray(facility))
+                facilitiesRelationsToSelectedHealthcareProfessional = facilitiesForRelationshipCreationArray
+                    .map(createFacilityRelation)
             }
+
             const updateHealthcareProfessionalInput: MutationUpdateHealthcareProfessionalArgs = {
                 id: selectedHealthcareProfessionalId.value,
                 input: {
                     acceptedInsurance: healthcareProfessionalSectionFields.acceptedInsurance,
                     degrees: healthcareProfessionalSectionFields.degrees,
-                    facilityIds: facilitiesRelationsToSelectedHealthcareProfessional.value.length
-                      > 0
-                        ? facilitiesRelationsToSelectedHealthcareProfessional.value
+                    facilityIds: facilitiesRelationsToSelectedHealthcareProfessional.length > 0
```

Comment:
tip: you can just check `.length` without the 0 check
```suggestion
                    facilityIds: facilitiesRelationsToSelectedHealthcareProfessional.length
```